src/main/java/org/truffleruby/core/binding/BindingNodes.java

```diff
@@ -175,10 +175,9 @@ public static boolean hiddenVariable(String name) {
         @Specialization
         public DynamicObject dup(DynamicObject binding) {
             MaterializedFrame frame = getTopFrame(binding);
-            Layouts.BINDING.setExtras(binding, newExtrasFrame(getContext(), frame));
             DynamicObject copy = allocateObjectNode.allocate(
                     Layouts.BASIC_OBJECT.getLogicalClass(binding),
-                    newExtrasFrame(getContext(), frame),
+                    frame,
                     null);
```

Comment:
Possibly, I'm actually wondering if `extras` can now be removed entirely. I know complications with eval and dup stopped me doing it before.